test/test_other.py

```diff
@@ -10112,6 +10112,9 @@ def print_percent(actual, expected):
       outputs += ['a.wasm']
       test_name += '_wasm'
 
+    if clear_cache:
+      self.run_process([EMCC, '--clear-cache'])
```

Comment:
You should not need to add this to the text code.   The cache should only need to be clear if things like system_libs.py changes.. it should never depend on the specific test being run.